File: taxee_server/taxee_server/Models/System.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations.Schema;

namespace taxee_server.Models
{
    [Index(nameof(name))]
    public class System
    {
        public long SystemID { get; set; }
        public string name { get; set; }
        public Coordinates coords { get; set; }
        public Body[] bodies { get; set; }
        public Station[] Stations { get; set; }

        [Microsoft.EntityFrameworkCore.Owned]
        public class Coordinates
        {
            public float x { get; set; }
            public float y { get; set; }
            public float z { get; set; }
        }
    }
}

File: taxee_client/web/models/system.dart
import 'station.dart';
import 'body.dart';

class System {
  String systemID, name;
  ReceivedCoordinates coordinates;
  double distance;
  Body? bodies;
  Station? stations;

  System({
    required this.systemID,
    required this.name,
    required this.coordinates,
    required this.distance,
    this.bodies,
    this.stations,
  });
}

class ReceivedCoordinates {
  double x, y, z;

  ReceivedCoordinates({
    required this.x,
    required this.y,
    required this.z,
  });
}
